python/sampling.cpp

```diff
@@ -122,24 +122,25 @@ std::string sample_str(const trace_entry& s) {
         return util::pprintf("<arbor.sample: time {} ms, \tvalue {}>", s.t, s.v);
 }
 
-void register_samples(pybind11::module& m) {
+void register_sampling(pybind11::module& m) {
     using namespace pybind11::literals;
 
-    // Samples
+    // Sample
     pybind11::class_<trace_entry> sample(m, "sample");
     sample
         .def_readonly("time", &trace_entry::t, "The sample time [ms] at a specific probe.")
         .def_readonly("value", &trace_entry::v, "The sample record at a specific probe.")
         .def("__str__",  &sample_str)
         .def("__repr__", &sample_str);
 
-    // Sample recorder
+    // Sampler
     pybind11::class_<sampler, std::shared_ptr<sampler>> samplerec(m, "sampler");
     samplerec
         .def(pybind11::init<>())
         .def("samples", &sampler::samples,
```

Comment:
It might be useful for an interface whereby requesting the samples for a pid would return a type that contains:
* the cell gid
* the probe lid
* the probe location
* the variable being probed ('voltage' or 'current')
* a list of the samples